src/Core/Grid/Query/Api/HookQueryBuilder.php

```diff
@@ -0,0 +1,148 @@
+     * @param DoctrineSearchCriteriaApplicator $searchCriteriaApplicator
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix,
+        private readonly DoctrineSearchCriteriaApplicator $searchCriteriaApplicator,
+    ) {
+        parent::__construct($connection, $dbPrefix);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getQueryBuilder($searchCriteria->getFilters());
+        $qb->select('h.id_hook, h.name, h.title, h.description, h.active, h.position');
+
+        $this->searchCriteriaApplicator
+            ->applyPagination($searchCriteria, $qb)
+            ->applySorting($searchCriteria, $qb);
+
+        return $qb;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getQueryBuilder($searchCriteria->getFilters());
+        $qb->select('COUNT(h.id_hook)');
+
+        return $qb;
+    }
+
+    /**
+     * Get generic query builder.
+     *
+     * @param array $filters
+     *
+     * @return QueryBuilder
+     */
+    private function getQueryBuilder(array $filters)
+    {
+        $qb = $this->connection
+            ->createQueryBuilder()
+            ->from($this->dbPrefix . 'hook', 'h');
+
+        foreach ($filters as $filterName => $filterValue) {
+            if ('id_hook' === $filterName) {
+                $qb->andWhere("h.id_hook = :$filterName");
+                $qb->setParameter($filterName, $filterValue);
+
+                continue;
+            }
+
+            if ('title' === $filterName) {
+                $qb->andWhere("h.title LIKE :$filterName");
+                $qb->setParameter($filterName, '%' . $filterValue . '%');
+
+                continue;
+            }
+
+            if ('name' === $filterName) {
+                $qb->andWhere("h.name LIKE :$filterName");
+                $qb->setParameter($filterName, '%' . $filterValue . '%');
+
+                continue;
+            }
+
+            if ('description' === $filterName) {
+                $qb->andWhere("h.description LIKE :$filterName");
+                $qb->setParameter($filterName, '%' . $filterValue . '%');
+
+                continue;
+            }
+
+            if ('position' === $filterName) {
+                // When filtering by position,
+                // value must be decreased by 1,
+                // since position value in database starts at 0,
+                // but for user display positions are increased by 1.
+                if (is_numeric($filterValue)) {
+                    --$filterValue;
+                } else {
+                    $filterValue = null;
+                }
+
+                $qb->andWhere("h.position = :$filterName");
+                $qb->setParameter($filterName, $filterValue);
+
+                continue;
+            }
```

Comment:
```suggestion
            if ('position' === $filterName && is_numeric($filterValue)) {
                // When filtering by position, value must be decreased by 1, since position value in database starts at 0
                // but for user display positions are increased by 1.
                $qb->andWhere("h.position = :$filterName");
                $qb->setParameter($filterName, $filterValue - 1);

                continue;
            }
```
I'm not at ease with decrementing the value provided when a simple subtraction can do the job and the code can be simplified and reduced